databricks/koalas/frame.py

```diff
@@ -6489,6 +6489,65 @@ def __class_getitem__(cls, params):
         # Koalas DataFrame.
         is_dataframe = None
 
+    def tail(self, n=5):
+        """
+        Return the last `n` rows.
+        This function returns last `n` rows from the object based on
+        position. It is useful for quickly verifying data, for example,
+        after sorting or appending rows.
+
+        Parameters
+        ----------
+        n : int, default 5
+            Number of rows to select.
+
+        Returns
+        -------
+        type of caller
+            The last `n` rows of the caller object.
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame({'animal':['alligator', 'bee', 'falcon', 'lion',
+        ...                    'monkey', 'parrot', 'shark', 'whale', 'zebra']})
+        >>> kdf
+              animal
+        0  alligator
+        1        bee
+        2     falcon
+        3       lion
+        4     monkey
+        5     parrot
+        6      shark
+        7      whale
+        8      zebra
+        Viewing the last 5 lines
+        >>> kdf.tail()
+           animal
+        0  monkey
+        1  parrot
+        2   shark
+        3   whale
+        4   zebra
+        Viewing the last `n` lines (three in this case)
+        >>> kdf.tail(3)
+           animal
+        0  shark
+        1  whale
+        2  zebra
+        """
+        # pick one column to perform order by
+        first_column = self.columns[0]
+
+        # calculate columns to be dropped
+        columns_to_drop = [col.name for col in self._sdf.schema if col.name not in self.columns]
+
+        # calculate last n records
+        data = self._sdf.orderBy(first_column).drop(*columns_to_drop).collect()[-n:]
```

Comment:
Actually this isn't quite performant. See https://github.com/databricks/koalas/issues/343#issuecomment-495416050 

If the DataFrame is big, it will collects all the data into the Spark's driver side (client) and it will easily hit an out-of-memory issue.

One possible way is that we collect tails from each partition, collect, and calculate the tail again.

One way might be to use `mapInPandas` which will be introduced in Spark 3.0 - https://github.com/apache/spark/blob/a5a5da78cfbb1b439c139fd2ffd5744fc60ebafe/python/pyspark/sql/functions.py#L3207-L3232

If we don't have a better idea now, let's leave this case closed for now.

Reply:
Yaa. Let me try that.

Reply:
If the rule is find the last n rows when sort by first column, then we can turn it into the operation "get first n rows", we only need to reverse the sorter compare function. What do you think ? @HyukjinKwon @akhilputhiry

Reply:
I am not getting time to work on this. Closing this for now.